src/instructlab/sdg/llmblock.py

```diff
@@ -84,27 +90,40 @@ def _parse(self, generated_string) -> dict:
 
         return matches
 
+    def _format_prompt(self, sample: Dict) -> str:
+        return self.prompt_template.format(**sample).strip()
+
     def _generate(self, samples, **gen_kwargs) -> list:
         prompts = [
-            self.model_prompt.format(
-                prompt=self.prompt_template.format(**sample).strip()
-            )
+            self.model_prompt.format(prompt=self._format_prompt(sample))
             for sample in samples
         ]
-        response = self.client.completions.create(
-            prompt=prompts, **{**self.defaults, **gen_kwargs}
-        )
-        return [choice.text.strip() for choice in response.choices]
+        generate_args = {**self.defaults, **gen_kwargs}
+
+        if self.server_supports_batched:
+            response = self.client.completions.create(
+                prompt=prompts, **generate_args
+            )
+            return [choice.text.strip() for choice in response.choices]
+
+        n = gen_kwargs.get("n", 1)
+        results = []
+        for prompt in prompts:
+            for _ in range(n):
+                response = self.client.completions.create(
+                    prompt=prompt, **generate_args
+                )
```

Comment:
We could rewrite this as 
```
responses = [
    self.client.completions.create(prompt=prompt, **generate_args)
    for prompt in prompts
    for _ in range(n)
]
```
wdyt?

Reply:
yes but then we would require an additional loop anyhow

Reply:
Yeah